moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1941,37 +1938,47 @@ double moveit::core::RobotState::computeCartesianPath(const JointModelGroup* gro
     if (setFromIK(group, pose, link->getName(), 1, 0.0, validCallback, options))
     {
       traj.push_back(RobotStatePtr(new RobotState(*this)));
-
-      // compute the distance to the previous point (infinity norm)
-      if (test_joint_space_jump)
-      {
-        double dist_prev_point = traj.back()->distance(*traj[traj.size() - 2], group);
-        dist_vector.push_back(dist_prev_point);
-        total_dist += dist_prev_point;
-      }
     }
     else
       break;
     last_valid_percentage = percentage;
   }
 
-  if (test_joint_space_jump && !dist_vector.empty())
+  if (test_joint_space_jump)
   {
-    // compute the average distance between the states we looked at
-    double thres = jump_threshold * (total_dist / (double)dist_vector.size());
-    for (std::size_t i = 0; i < dist_vector.size(); ++i)
-      if (dist_vector[i] > thres)
-      {
-        logDebug("Truncating Cartesian path due to detected jump in joint-space distance");
-        last_valid_percentage = (double)i / (double)steps;
-        traj.resize(i);
-        break;
-      }
+    last_valid_percentage *= testJointSpaceJump(group, traj, jump_threshold);
```

Comment:
Let's say 15 of 20 steps were succesfull, so last_valid_percentage would be 0.75. If testJointSpace now returns 1.0 because there are no joint space jumps in these 15 steps, we still want to return 0.75. If testJointSpace fails after half of the steps and returns 0.5 we want to return 0.75 * 0.5;